email/src/main/java/edu/jhuapl/dorset/demos/EmailManager.java

```diff
@@ -0,0 +1,308 @@
+        }
+    }
+    
+    /**
+     * Get an email from a folder
+     * 
+     * @param folder   the folder to retrieve an email from
+     * @return the email
+     * @throws MessagingException   if email cannot be retrieved
+     */
+    public Message getMessage(FolderType folder) throws MessagingException {
+        try {
+            return getFolder(folder).getMessage(1);
+        } catch (MessagingException e) {
+            close();
+            throw new MessagingException("Failed to retrieve email from " + folder + "folder", e);
+        }  
+    }
+    
+    /**
+     * Mark an email as seen
+     * 
+     * @param msg  the email to be marked seen
+     * @throws MessagingException   if email cannot be properly marked
+     */
+    //not currently being used, but may be needed for threading
+    public void markSeen(Message msg) throws MessagingException {
+        try {
+            msg.setFlag(Flags.Flag.SEEN, true);
+        } catch (MessagingException e) {
+            throw new MessagingException("Failed to mark email as seen", e);
+        }
+    }
+
+    /**
+     * Retrieve and return text from an email
+     * 
+     * @param msg   the email to be read
+     * @return the text the email
+     * @throws MessagingException   if email cannot be accessed
+     */
+    public String readEmail(Message msg) throws MessagingException {
+        return getText(msg);
+    }
+
+    /**
+     * Get the text of an email
+     * 
+     * @param part   the email body to be retrieved
+     * @return text  the text of the email
+     * @throws MessagingException   if email body cannot be retrieved
+     */
+    private String getText(Part part) throws MessagingException {
+        try {
+            String text = "";
+            if (part.isMimeType(TEXT_PLAIN)) {
+                text += ((String) part.getContent());
+            } else if (part.isMimeType(MULTIPART)) {
+                Multipart mp = (Multipart) part.getContent();
+                int count = mp.getCount();
+                for (int n = 0; n < count; n++) {
+                    text += (getText(mp.getBodyPart(n)));
+                }
+            } else if (part.isMimeType(ENCAPSULATED)) {
+                text += getText((Part) part.getContent());
+            }
+            logger.info("email reads: " + text);
+            return text;
+        } catch (MessagingException | IOException e) {
+            throw new MessagingException("Failed to read email body", e);
+        }
+    }
+
+    /**
+     * Send a reply to the specified email
+     *
+     * @param response   the text to be used for the body of the reply
+     * @param msg  the email to be responded to
+     * @throws MessagingException   if rely cannot be sent
+     */
+    public void sendMessage(String response, Message msg) throws MessagingException {
+        try {
+            logger.info("response reads: " + response);
+            Message replymsg = new MimeMessage(session);
```

Comment:
Also you set the variable again in the next line. I'm confused why both are there.